src/main/java/ca/on/oicr/gsi/dimsum/util/reporting/reports/DonorAssayReport.java

```diff
@@ -1,43 +1,65 @@
 import ca.on.oicr.gsi.dimsum.util.reporting.Column;
 import ca.on.oicr.gsi.dimsum.util.reporting.Report;
 import ca.on.oicr.gsi.dimsum.util.reporting.ReportSection;
 import ca.on.oicr.gsi.dimsum.util.reporting.ReportSection.TableReportSection;
-import ca.on.oicr.gsi.dimsum.util.reporting.reports.shared.CaseSampleRowData;
 
 public class DonorAssayReport extends Report {
 
-  private static final ReportSection<CaseSampleRowData> mainSection =
-      new TableReportSection<CaseSampleRowData>("Donor Assay Report", Arrays.asList(
-          Column.forString("Project", x -> x.getSample().getProject()),
+  private static class RowData {
+
+    private final Case kase;
+
+    public RowData(Case kase, Test test, Assay assay) {
+      this.kase = kase;
+    }
+
+    public Case getCase() {
+      return kase;
+    }
+
+  }
+
+  private static final ReportSection<RowData> mainSection =
+      new TableReportSection<RowData>("Donor Assay Report", Arrays.asList(
+          Column.forString("Project", x -> x.getCase().getProjects().stream()
+              .map(Project::getName)
+              .collect(Collectors.joining(", "))),
           Column.forString("Pipeline", x -> getProjectPipelines(x.getCase())),
           Column.forString("Donor", x -> x.getCase().getDonor().getName()),
           Column.forString("External Name", x -> x.getCase().getDonor().getExternalName()),
           Column.forString("Assay", x -> x.getCase().getAssayName()),
-          Column.forString("Start Date", x -> getStartDate(x.getCase()).toString()),
-          Column.forString("Latest Activity", x -> getLatestActivityDate(x.getCase()).toString()),
+          Column.forString("Start Date", x -> x.getCase().getStartDate().toString()),
+          Column.forString("Latest Activity", x -> x.getCase().getLatestActivityDate().toString()),
           Column.forString("Status", x -> getCaseStatus(x.getCase())),
           Column.forString("Stop/Pause Reason", x -> getStopPauseReason(x.getCase())))) {
         @Override
-        public List<CaseSampleRowData> getData(CaseService caseService, JsonNode parameters) {
+        public List<RowData> getData(CaseService caseService, JsonNode parameters) {
           Set<String> caseIds = getParameterStringSet(parameters, "caseIds");
-          if (caseIds == null) {
-            throw new BadRequestException("caseIds parameter missing");
+          if (caseIds == null || caseIds.isEmpty()) {
+            throw new BadRequestException("caseIds parameter missing or empty");
           }
-          return CaseSampleRowData.listByCaseIds(caseService, caseIds);
+          List<Case> cases = caseService.getCasesByIds(caseIds);
+          Map<Long, Assay> assaysById = caseService.getAssaysById();
+          return cases.stream()
+              .flatMap(kase -> kase.getTests().stream()
+                  .map(test -> new RowData(kase, test, assaysById.get(kase.getAssayId()))))
+              .collect(Collectors.toList());
```

Comment:
Now the report will create duplicate rows for every test in a case, and you're not even using the test or assay that you feed to the RowData constructor. No transformation is necessary, and the RowData class is not necessary. This should simply return the list of cases.